drivers/dht/dht.c

```diff
@@ -19,192 +20,318 @@
  * @author      Ludwig Knüpfer <ludwig.knuepfer@fu-berlin.de>
  * @author      Christian Mehlis <mehlis@inf.fu-berlin.de>
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Hugues Larrive <hugues.larrive@pm.me>
  *
  * @}
  */
 
+#include <errno.h>
 #include <stdint.h>
 #include <string.h>
 
-#include "log.h"
 #include "assert.h"
-#include "xtimer.h"
-#include "timex.h"
-#include "periph/gpio.h"
-
 #include "dht.h"
 #include "dht_params.h"
+#include "log.h"
+#include "periph/gpio.h"
+#include "time_units.h"
+#include "ztimer.h"
+
+#if __AVR__
+/* On AVR, `gpio_read()` takes 2 µs mainly due to the port register
+ * 16-bit address computation, so we will need those libs to compute it
+ * once and for all. */
```

Comment:
In fact, the slowness of gpio_read() (2 µs) wasn't really a problem here. It just tripled the duration of the iteration, which was a bit surprising but not really a drawback. It can work without any issues up to 10 or 12 µs per iteration."